### PR TITLE
Add confirmation when publish without deployments

### PR DESCRIPTION
**Pull request by Krishanx92**
Screenshot from 2021-04-12 09-17-36

Fix https://github.com/wso2/product-apim/issues/10941

### CODE REVIEW

**Comment by erangatl** [comment]
Shall we remove the exclamation mark?

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx [hunk 1]
```diff
@@ -288,6 +344,70 @@ class LifeCycleUpdate extends Component {
...
+                            id='Apis.Details.LifeCycle.components'
+                            defaultMessage='Publish without deploying'
+                        />
+                    </DialogTitle>
+                    <DialogContent>
+                        <DialogContentText id='alert-dialog-description'>
+                            <Typography variant='subtitle1' display='block' gutterBottom>
+                                <FormattedMessage
+                                    id='Apis.Details.LifeCycle.publish.content'
+                                    defaultMessage='Publish API without deployments will affect API runtime'
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx:364]:
```suggestion
                                    defaultMessage='The API will not be available for consumption unless it is deployed.'
```

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx [hunk 2]
```diff
@@ -288,6 +344,70 @@ class LifeCycleUpdate extends Component {
...
+                <Dialog
+                    open={isOpen}
+                    onClose={() => this.setIsOpen(false)}
+                    aria-labelledby='alert-dialog-title'
+                    aria-describedby='alert-dialog-description'
+                >
+                    <DialogTitle id='alert-dialog-title'>
+                        <FormattedMessage
+                            id='Apis.Details.LifeCycle.components'
+                            defaultMessage='Publish without deploying'
```
Comment (malinthaprasan) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx:356]:
```suggestion
                            defaultMessage='Publish API without deployments'
```

**Comment by malinthaprasan** [comment]
#### Publish API without deployments

##### The API will not be available for consumption unless it is deployed.
###### Are you sure you want to continue?

`Cancel`         `Yes, Publish API`       `Deployments [>]`

**Comment by malinthaprasan** [comment]
Will something like above do? 
@Krishanx92 @tmkasun

**Comment by erangatl** [comment]
> ###### Are you sure you want to continue?

We should avoid ambiguous questions. Shall we remove this?
http://wso2.github.io/carbon-apimgt/HIG/#dialogs

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx (hunk 1 above)
Comment (Krishanx92) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx]: Fixed.

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx (hunk 2 above)
Comment (Krishanx92) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/LifeCycle/LifeCycleUpdate.jsx]: Fixed.

**Review by tmkasun — APPROVED** [review]

**Pull request merged**